src/build/properties.ts

```diff
@@ -1,22 +1,40 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
-import { FullResolution, ThemePreset } from '../shared/theme';
+import { FullResolution, Theme, ThemePreset, resolveContext, resolveTheme } from '../shared/theme';
 import { toCssVarName } from './token';
 
-/**
- * Generates a map between design tokens and CSS Custom properties for the past theme.
- * The optional list of variables will override any generated names
- *
- * @param theme
- * @returns map between design tokens and CSS custom properties
- */
+interface TokensValuesMap {
+  [token: string]: Array<string>;
+}
+
 export function calculatePropertiesMap(
-  resolution: FullResolution,
+  themes: Array<Theme>,
   variablesMap: ThemePreset['variablesMap']
 ): Record<string, string> {
-  const mapEntries = Object.entries(resolution).map(([token, assignment]) => {
-    const values = typeof assignment === 'string' ? [assignment] : Object.values(assignment);
+  const resolutions: Array<FullResolution> = [];
+  themes.forEach((theme) => {
+    resolutions.push(resolveTheme(theme));
+    Object.values(theme.contexts || {}).forEach((context) => {
+      resolutions.push(resolveContext(theme, context));
+    });
+  });
+
+  const tokensValuesMap = getTokensValuesFromResolutions(resolutions);
+
+  const mapEntries = Object.entries(tokensValuesMap).map(([token, values]) => {
     return [token, toCssVarName(variablesMap[token], values.filter((value) => !!value) as Array<string>)];
   });
   return Object.fromEntries(mapEntries);
 }
+
+const getTokensValuesFromResolutions = (resolutions: Array<FullResolution>): TokensValuesMap =>
+  resolutions.reduce((tokensValuesMap: TokensValuesMap, resolution: FullResolution) => {
+    Object.entries(resolution).map(([token, assignment]) => {
+      const values = typeof assignment === 'string' ? [assignment] : Object.values(assignment);
+      if (!tokensValuesMap[token]) {
+        tokensValuesMap[token] = [];
+      }
+      tokensValuesMap[token] = [...tokensValuesMap[token], ...values];
```

Comment:
Here we're modifying the `tokensValuesMap` directly. I'm not very familiar with the package, but is there a risk of an unwanted side effect if this is used somewhere else?

Alternatively, we could create a new object, modify and return it instead.
```typescript
const newTokensValuesMap = { ...tokensValuesMap };
```

Reply:
The `tokensValuesMap ` is initialized in the reduce loop within this function, so it is not used elsewhere.

Reply:
Ah, yes. I was looking at this function in isolation. LGTM!